src/azure-cli/azure/cli/command_modules/vm/operations/vmss.py

```diff
@@ -0,0 +1,27 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=no-self-use, line-too-long, protected-access, too-few-public-methods, unused-argument
+from knack.log import get_logger
+
+from ..aaz.latest.vmss import ListInstances as _VMSSListInstances
+
+logger = get_logger(__name__)
+
+
+class VMSSListInstances(_VMSSListInstances):
+    def _output(self, *args, **kwargs):
+        from azure.cli.core.aaz import AAZUndefined, has_value
+
+        # Resolve flatten conflict
+        # When the type field conflicts, the type in inner layer is ignored and the outer layer is applied
+        for value in self.ctx.vars.instance.value:
+            if has_value(value.resources):
+                for resource in value.resources:
+                    if has_value(resource.type):
+                        resource.type = AAZUndefined
```

Comment:
Why do we need this logic?

Reply:
It is moved from custom.py file and not related to this PR. It was imported by this PR: https://github.com/Azure/azure-cli/pull/27128
I remembered that this was caused by a conflict in the output field in swagger. Had a discussion with Kai at the time and this was the best solution.